daemon/api_validate.go

```diff
@@ -289,3 +313,55 @@ func forgetValidationSet(st *state.State, accountID, name string, sequence int)
 	assertstate.DeleteValidationSet(st, accountID, name)
 	return SyncResponse(nil, nil)
 }
+
+func validateAgainstStore(st *state.State, c *Command, accountID, name string, sequence int, user *auth.UserState) Response {
+	// not available locally, try to find in the store.
+	as, err := getSingleSeqFormingAssertion(c, accountID, name, sequence, user)
+	if _, ok := err.(*asserts.NotFoundError); ok {
+		return validationSetNotFound(accountID, name, sequence)
+	}
+	if err != nil {
+		return InternalError(err.Error())
+	}
+	sets := snapasserts.NewValidationSets()
+	vset := as.(*asserts.ValidationSet)
+	if err := sets.Add(vset); err != nil {
+		return InternalError(err.Error())
+	}
+	snaps, err := installedSnaps(st)
+	if err != nil {
+		return InternalError(err.Error())
+	}
+
+	validErr := checkInstalledSnaps(sets, snaps)
+	res := validationSetResult{
+		AccountID: vset.AccountID(),
+		Name:      vset.Name(),
+		Sequence:  vset.Sequence(),
+		// XXX: pass actual err details and implement "verbose" mode
+		// for the client?
```

Comment:
let's leave a TODO here and consider it again once we have implemented a bit more of the rest